### PR TITLE
Feat: Change Workplace ID

### PR DESCRIPTION
**Pull request by chadtomkiss-madetech**
#### Issue
Admins would like the ability to change the Workplace ID when approving registrations

#### Work done

- Moved registrations into their own components
- Updated `/api/admin/approval/` endpoint to accept an nmdsId and also check it's unique
- Added a `firstErrorPipe` to get the first error for a form control (so that you don't display multiple errors under an input)
- Added `testing-library/angular` to easily interact with Angular components in tests
- Added tests

#### Tests
Does this PR include tests for the changes introduced?
- [ ] Yes
- [x] Yes but could do with more
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -9,8 +9,6 @@ import { NestedRoutesService } from '@core/services/nested-routes.service';
 import { Angulartics2GoogleTagManager } from 'angulartics2/gtm';
 import { filter, take, takeWhile } from 'rxjs/operators';
 
-enableProdMode();
```
Comment (chadtomkiss-madetech) [inline src/app/app.component.ts:12]: Having `enableProdMode()` strips some keys from objects that's needed for running tests

i.e `fixture.debugElement`

I've taken this out but may need to wrap in an `if` to only enable when not running the tests

File: server/routes/admin/approval/index.js [hunk 2]
```diff
@@ -84,16 +85,35 @@ router.route('/').post(async (req, res) => {
       console.log(error);
     }
   }else{
+    const nmdsId = req.body.nmdsId;
```
Comment (chadtomkiss-madetech) [inline server/routes/admin/approval/index.js:88]: This endpoint is missing tests

File: src/app/features/search/registrations/registrations.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    public registrationsService: RegistrationsService,
+  ) {}
+
+  ngOnInit() {
+    this.getRegistrations();
+  }
+  public getRegistrations() {
+    this.registrationsService.getRegistrations().subscribe(
+      data => {
+        this.registrations = data;
```
Comment (chadtomkiss-madetech) [inline src/app/features/search/registrations/registrations.component.ts:21]: I had trouble testing Observables - I think that the `subscribe` isn't being run when first rendering a component.

File: src/app/features/search/registrations/registrations.component.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      },
+      error => this.onError(error)
+    );
+  }
+  public handleRegistration(index: number) {
```
Comment (chadtomkiss-madetech) [inline src/app/features/search/registrations/registrations.component.ts:26]: Fix spacing

File: server/routes/admin/approval/index.js [hunk 5]
```diff
@@ -2,10 +2,11 @@
 const express = require('express');
 const router = express.Router();
 const models = require('../../../models');
+const Sequelize = require('sequelize');
+const { ne } = Sequelize.Op;
```
Comment (aaron-russell) [inline server/routes/admin/approval/index.js:6]: Does this name help identify it in the future?

File: src/app/app.component.ts (hunk 1 above)
Comment (aaron-russell) [reply src/app/app.component.ts:12]: https://stackoverflow.com/a/39678498 you can enable it in certain environments

**Comment by aaron-russell** [comment]
You may also want to connect a Trello ticket

**Comment by chadtomkiss-madetech** [comment]
[Change WorkplaceID on Registration approval](https://trello.com/c/CdH9q1rN/40-change-workplaceid-on-registration-approval)

File: server/routes/admin/approval/index.js (hunk 5 above)
Comment (chadtomkiss-madetech) [reply server/routes/admin/approval/index.js:6]: Thanks for questioning this!

It's a "not equal" operator and the naming was based on the docs

Do you think it needs a comment/link to the docs for a list of operators?

File: src/app/app.component.ts (hunk 1 above)
Comment (chadtomkiss-madetech) [reply src/app/app.component.ts:12]: I have added a `environment.test.ts` and set the default `environment.ts` to true.

File: server/routes/admin/approval/index.js (hunk 5 above)
Comment (aaron-russell) [reply server/routes/admin/approval/index.js:6]: Yeah, makes sense, as reading that, I wasn't sure what ne stood for. THing we have used `op` elsewhere but I don't feel this is any better! :/

**Pull request merged**